lib/composer.js

```diff
@@ -8,13 +8,28 @@ function loadJsonFile(basePath, fileName) {
   return JSON.parse(loadedFile);
 }
 
-function getFirstAvailableAliasVersion(applicationDataObj) {
+function getAliasVersion(applicationDataObj, versionFound) {
+  // if the version is an alias, we look it up
   var availableAliases = _.get(applicationDataObj, 'extra[\'branch-alias\']');
   if (availableAliases) {
-    return _.get(availableAliases, Object.keys(availableAliases)[0]);
+    // we try for a matching alias, otherwise the first found
+    return _.get(availableAliases, versionFound,
+      availableAliases[Object.keys(availableAliases)[0]]);
   }
 }
 
+function getVersion(applicationDataObj) {
+  var versionFound = _.get(applicationDataObj, 'version');
+  return (versionFound && /\d/.test(versionFound)) ? versionFound :
```

Comment:
What does `/\d/` mean in this context? I think it'll match any of the following `versionFound` values:
`1.0.0`
`abc1def`
`something0`

What is it aiming to match?

Reply:
If this is matching aliases, do we expect aliases to have no digits?

Reply:
the version specified in the `version` property can be either an actual version or be an alias, in which case it should be defined under `extra.branch-alias`. When we look there, it's not so simple either. It can be an inline alias as versions, e.g `1.6 as 1.7`(`alias` as `version`), or it can even be one alias as another etc. 
It can be a reference to a branch name in some external repository, if a `repositories` property exists, and to really be sure, we may need to actually pull that repo and read its' info.
For now, I decided to make an assumption that aliases don't have digits in them. I agree these issues need to be addressed.
I already started working on this in #5, and I'll would appreciate hearing your thoughts on prioritization here.

Reply:
Ok! Since this is an assumption, why not skip it altogether and do the following:
- extract version string
- try to see if it matches any alias
  - if so, use the aliased string
  - if no, stay with what was found initially

This way this assumption is not needed.

Reply:
Because `otherwise the first found` on line 15 also seems like an assumption to me... Correct?

Reply:
@adrukh 
Many times there is no `version` property, so nothing to compare the version to, but the actual version is usually the first property of `extra.branch-alias`.
So I suggest we try the following (adding on what you wrote eariler):

* extract version string
* if exists, try to see if it matches any alias
    * if so, use the aliased string
    * if no, fallback in the following order
        1. use version if found
        2. use the first alias found in `extra.branch-alias`.

Reply:
👍

Reply:
@anton updated